Issue during cross compilation of open-vcdiff for powerpc-linux

Original [issue 38](https://code.google.com/p/open-vcdiff/issues/detail?id=38) created by suyash.kools on 2013-01-03T05:33:47.000Z:

<b>What steps will reproduce the problem?</b>
1. Configure open-vcdiff to be compiled for powerpc by using `--host=powerpc-linux` option with ./configure in the spec file.
2. start the compilation by going into the root directory of your linux platform powerpc project and use the commands:
```sh
make -C build open-vcdiff.addpkg
make -C build open-vcdiff
```
3. The compilation starts but throws a compilation error.


<b>What is the expected output? What do you see instead?</b>
The compilation is supposed to complete and produce a cross compiled RPM for powerpc. Instead we see error like this:
```
libvcdcom.so.0, needed by ./.libs/libvcddec.so, not found (try using -rpath or -rpath-link)
./.libs/libvcddec.so: undefined reference to `open_vcdiff::g_fatal_error_occurred'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VarintBE&lt;long long&gt;::Parse(char const*, char const**)'
./.libs/libvcdenc.so: undefined reference to `open_vcdiff::VarintBE&lt;int&gt;::Length(int)'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VCDiffCodeTableData::Validate(unsigned char) const'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VCDiffAddressCache::Init()'
./.libs/libvcdenc.so: undefined reference to `open_vcdiff::VarintBE&lt;int&gt;::AppendToOutputString(int, open_vcdiff::OutputStringInterface*)'
./.libs/libvcdenc.so: undefined reference to `open_vcdiff::VarintBE&lt;long long&gt;::Length(long long)'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VarintBE&lt;int&gt;::Parse(char const*, char const**)'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VCDiffAddressCache::DecodeAddress(int, unsigned char, char const**, char const*)'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VCDiffAddressCache::VCDiffAddressCache()'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VCDiffCodeTableData::kDefaultCodeTableData'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VCDiffInstructionName(open_vcdiff::VCDiffInstructionType)'
./.libs/libvcddec.so: undefined reference to `open_vcdiff::VCDiffAddressCache::VCDiffAddressCache(int, int)'
./.libs/libvcdenc.so: undefined reference to `open_vcdiff::VCDiffAddressCache::EncodeAddress(int, int, int*)'
./.libs/libvcddec.so: undefined reference to `adler32'
./.libs/libvcdenc.so: undefined reference to `open_vcdiff::VarintBE&lt;long long&gt;::AppendToOutputString(long long, open_vcdiff::OutputStringInterface*)'
./.libs/libvcdenc.so: undefined reference to `open_vcdiff::VarintBE&lt;int&gt;::AppendToString(int, std::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt;*)'
collect2: ld returned 1 exit status
Makefile:836: *** [vcdiff] Error 1
```
<b>What version of the product are you using? On what operating system?</b>

I am using open-vcdiff-0.8.3 compiling for WindRiver linux for powerpc architecture.

<b>Please provide any additional information below.</b>
I am guessing that it can be result of improperly defined macros in the programs where the above function calls cannot be made because maybe they are not defined for powerpc. I will have to look up the code for this. If anyone has faced a similar issue Please help me with this. I am stuck without a way forward.

--- a/src/headerparser.cc
+++ b/src/headerparser.cc
@@ -14,17 +14,15 @@
 
 #include "headerparser.h"
 
-#include <limits.h>
+#include <limits>
 
-#include <config.h>
 #include "logging.h"
 #include "varint_bigendian.h"
-#include "vcdiff_defs.h"
 
 namespace {
 
 bool SumWouldOverflow2(size_t a, size_t b) {
-  return a > SIZE_MAX - b;
+  return a > std::numeric_limits<size_t>::max() - b;
 }
 
 bool SumWouldOverflow4(size_t a, size_t b, size_t c, size_t d) {

